File: 222-count-complete-tree-nodes/222-count-complete-tree-nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def countNodes(self, root: Optional[TreeNode]) -> int:
        if root == None:
            return 0
        
        count = 0
        queue = deque()
        
        queue.append(root)
        
        while queue:
            currNode = queue.popleft()
            count += 1
            
            if currNode.left != None:
                queue.append(currNode.left)
            
            if currNode.right != None:
                queue.append(currNode.right)
                
        return count

File: 222-count-complete-tree-nodes/222-count-complete-tree-nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int countNodes(TreeNode* root) {
        return solve(root);
    }
private:
    int solve(TreeNode* root) {
        if(root == nullptr)
            return 0;
        
        queue<TreeNode*> queue;
        queue.push(root);
        int count = 0;
        
        while(!queue.empty()) {
            auto currNode = queue.front();
            queue.pop();
            
            ++count;
            if(currNode -> left) queue.push(currNode -> left);
            if(currNode -> right) queue.push(currNode -> right);
        }
        return count;
    }
};